droidlet/tools/crowdsourcing/servermgr/allocate_instances.py

```diff
@@ -0,0 +1,153 @@
+                        {"name": "TIMESTAMP", "value": timestamp},
+                        {
+                            "name": "SENTRY_DSN",
+                            "value": os.environ.get("CRAFTASSIST_SENTRY_DSN", ""),
+                        },
+                        {"name": "CLOUDFLARE_TOKEN", "value": os.getenv("CLOUDFLARE_TOKEN")},
+                        {"name": "CLOUDFLARE_ZONE_ID", "value": os.getenv("CLOUDFLARE_ZONE_ID")},
+                    ],
+                }
+            ]
+        },
+    )
+    logging.info("Launched: {}".format(r))
+    return r["tasks"][0]["taskArn"], timestamp
+
+
+def is_instance_up(instance_id):
+    try:
+        x = ecs.describe_tasks(cluster="craftassist", tasks=[instance_id])
+
+        attachment_id = x["tasks"][0]["containers"][0]["networkInterfaces"][0]["attachmentId"]
+        attachment = next(y for y in x["tasks"][0]["attachments"] if y["id"] == attachment_id)
+
+        eni = next(y for y in attachment["details"] if y["name"] == "networkInterfaceId")["value"]
+        ip = ec2.NetworkInterface(eni).private_ip_addresses[0]["Association"]["PublicIp"]
+
+        s = socket.socket()
+        s.settimeout(10)
+        s.connect((ip, 25565))
+        s.close()
+
+        ping_cuberite.ping(ip, 25565, timeout=1)
+
+    except:
+        return False
+
+    return True
+
+
+def get_instance_ip(instance_id):
+    try:
+        x = ecs.describe_tasks(cluster="craftassist", tasks=[instance_id])
+
+        attachment_id = x["tasks"][0]["containers"][0]["networkInterfaces"][0]["attachmentId"]
+        attachment = next(y for y in x["tasks"][0]["attachments"] if y["id"] == attachment_id)
+
+        eni = next(y for y in attachment["details"] if y["name"] == "networkInterfaceId")["value"]
+        ip = ec2.NetworkInterface(eni).private_ip_addresses[0]["Association"]["PublicIp"]
+    except:
+        raise ValueError(f'This instance {instance_id} should have been up, but failed to get ip')
+    return ip
+
+    
+def request_instance(instance_num):
+    logging.info(f"Requesting {instance_num} instances from AWS")
+    instances = [launch_instance(task="craftassist", config="flat_world", debug=False)[0] for _ in range(instance_num)]
+    instance_status = [False] * instance_num
+
+    while not all(instance_status):
+        time.sleep(5)
+        logging.info(f"Checking status of {instance_num} instances... Not ready!")
+        for i in range(instance_num):
+            instance_status[i] = is_instance_up(instances[i])
+    
+    instance_ips = [get_instance_ip(instance) for instance in instances]
+    logging.info(f"All {instance_num} are up, ip list: {instance_ips}")
+    return instance_ips
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--instance_num", type=int, default=1, help="number of instances requested")
+    args = parser.parse_args()
+    instance_ips = request_instance(args.instance_num)
```

Comment:
Can we write this to a file, so that it can be read by task scripts?

Reply:
oops saw it after merge. I think we can chain it with cloudflare&mephisto so that we can get a config file at the end (saving the trouble of storing IPs to a file and read it). wdyt?